src/mongoose_rabbit_worker.erl

```diff
@@ -185,44 +145,19 @@ publish(Exchange, RoutingKey, Message, Opts = #state{host = Host}) ->
                          [])
     end.
 
--spec user_topic_routing_key(JID :: {binary(), binary(), binary()},
-                             Topic :: binary()) -> binary().
-user_topic_routing_key({User, Host, _Res}, Topic) ->
-    JID = jid:to_binary({User, Host}),
-    <<JID/binary, ".", Topic/binary>>.
-
--spec make_presence_msg(JID :: binary(), Status :: atom()) -> binary().
-make_presence_msg(JID, Status) ->
-    Msg = #{user_id => jid:to_binary(JID), present => is_user_online(Status)},
-    jiffy:encode(Msg).
-
--spec make_chat_msg(From :: binary(), To :: atom(), UserMsg :: binary()) ->
-    binary().
-make_chat_msg(From, To, UserMsg) ->
-    Msg = #{to_user_id => jid:to_binary(To),
-            message => UserMsg,
-            from_user_id => jid:to_binary(From)},
-    jiffy:encode(Msg).
-
--spec is_user_online(online | offline) -> boolean().
-is_user_online(online) -> true;
-is_user_online(offline) -> false.
-
--spec publish_message_and_wait_for_confirm(Exchange :: binary(),
-                                           RoutingKey :: binary(),
-                                           Message :: binary(),
-                                           Opts :: publish_opts()) -> term().
-publish_message_and_wait_for_confirm(Exchange, RoutingKey, Message,
+-spec publish_message_and_wait_for_confirm(Method :: mongoose_amqp:method(),
+                                           Payload :: mongoose_amqp:message(),
+                                           worker_opts()) ->
+    true | no_return().
+publish_message_and_wait_for_confirm(Method, Payload,
                                      #state{channel = Channel,
-                                            confirms_enabled = IsConfirmOn}) ->
-    amqp_channel:call(Channel, #'basic.publish'{exchange = Exchange,
-                                                routing_key = RoutingKey},
-                      #amqp_msg{payload = Message}),
-    case IsConfirmOn of
+                                            confirms = IsConfirmEnabled}) ->
+    amqp_channel:call(Channel, Method, Payload),
```

Comment:
We are not handling the failures here, which may lead to loosing messages that are waiting in the process' message queue to be published.
I think we should apply similar retry mechanisms, which I described for `handle_amqp_call` function.